src/D2L.Bmx/Okta/OktaApi.cs

```diff
@@ -0,0 +1,111 @@
+using System.Net;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using D2L.Bmx.Okta.Models;
+namespace D2L.Bmx.Okta;
+
+internal interface IOktaApi {
+	void SetOrganization( string organization );
+	void AddSession( string sessionId );
+	Task<AuthenticateResponseInital> AuthenticateOktaAsync( AuthenticateOptions authOptions );
+	Task<AuthenticateResponseSuccess> AuthenticateChallengeMfaOktaAsync( AuthenticateChallengeMfaOptions authOptions );
+	Task<OktaSession> CreateSessionOktaAsync( SessionOptions sessionOptions );
+	Task<OktaApp[]> GetAccountsOktaAsync( string userId );
+	Task<string> GetAccountOktaAsync( Uri linkUri );
+}
+
+[JsonSourceGenerationOptions( PropertyNamingPolicy = JsonKnownNamingPolicy.CamelCase )]
+[JsonSerializable( typeof( AuthenticateOptions ) )]
+[JsonSerializable( typeof( AuthenticateChallengeMfaOptions ) )]
+[JsonSerializable( typeof( SessionOptions ) )]
+[JsonSerializable( typeof( AuthenticateResponseInital ) )]
+[JsonSerializable( typeof( AuthenticateResponseSuccess ) )]
+[JsonSerializable( typeof( OktaSession ) )]
+[JsonSerializable( typeof( OktaApp[] ) )]
+internal partial class SourceGenerationContext : JsonSerializerContext {
+}
+
+internal class OktaApi : IOktaApi {
+
+	private readonly CookieContainer _cookieContainer;
+	private readonly HttpClient _httpClient;
+
+	public OktaApi() {
+		_cookieContainer = new CookieContainer();
+		_httpClient = new HttpClient( new HttpClientHandler { CookieContainer = _cookieContainer } );
+		_httpClient.Timeout = TimeSpan.FromSeconds( 30 );
+		_httpClient.DefaultRequestHeaders.Accept.Add( new MediaTypeWithQualityHeaderValue( "application/json" ) );
+	}
+
+	public void SetOrganization( string organization ) {
```

Comment:
```suggestion
	public void IOktaApi.SetOrganization( string organization ) {
```

I think [this](https://github.com/Brightspace/instance-metadata-store/pull/643#discussion_r1124738319) could apply here?

Reply:
🤝